src/common/mailing.ts

```diff
@@ -69,36 +75,47 @@ export const resubscribeUser = async (
 };
 
 export const syncSubscription = async function (
-  userId: string,
+  userIds: string[],
   con: DataSource,
 ): Promise<void> {
   if (!process.env.CIO_APP_KEY) {
     return;
   }
 
-  const atts = await cioApi.getAttributes(userId);
-  const subs = JSON.parse(
-    atts?.customer?.attributes?.cio_subscription_preferences || '{}',
+  // This is fetched from APIClient source code
+  // TODO: Remove this once APIClient supports fetching attributes in bulk
+  const request = new CIORequest(process.env.CIO_APP_KEY, {});
+  const userAttributes = await request.post(
+    `${RegionUS.apiUrl}/customers/attributes`,
+    { ids: userIds },
   );
-  const unsubscribed = atts?.customer?.unsubscribed;
-  const marketing =
-    isSubscribed(subs, CioUnsubscribeTopic.Marketing) && !unsubscribed;
-  const notifications =
-    isSubscribed(subs, CioUnsubscribeTopic.Notifications) && !unsubscribed;
-  const digest =
-    isSubscribed(subs, CioUnsubscribeTopic.Digest) && !unsubscribed;
+
   await con.transaction(async (manager) => {
-    await manager
-      .getRepository(User)
-      .update(
-        { id: userId },
-        { notificationEmail: notifications, acceptedMarketing: marketing },
+    userAttributes?.customers.forEach(async (customer) => {
```

Comment:
Was a bit unsure what best practice would be here. I couldn't find any way of bulk updating users via TypeORM, and would probably have to use raw SQL for it.
So I went with the easiest solution, forEach loop inside the transaction.

Reply:
Thats fine, just see if we need to batch (as we do in scripts) so we don't have a large transaction that is gonna clog other updates.

Reply:
At most, it'll only do 100 updates, as that's the max length of the userIds list (and response from customer.io)

Reply:
Ok, note it in the comment above code and link docs just so next person is aware 👍

Reply:
Done.